### PR TITLE
Remove existing attribute from autocomplete suggestions

### LINKED ISSUE #80: Remove existing element attributes from the autocomplete list
Based on the [HTML standards](https://html.spec.whatwg.org/multipage/syntax.html#attributes-2), it is not valid to have duplicate attributes on an element.  The existing attribute should not appear in the autocomplete list.

Example:

image

### PR DESCRIPTION
**Pull request by sapphi-red**
This PR closes #80.

### CODE REVIEW

File: src/test/completion.test.ts [hunk 1]
```diff
@@ -69,15 +69,13 @@ suite('HTML Completion', () => {
 
 		testCompletionFor('<input t|ype="text"', {
 			items: [
-				{ label: 'type', resultText: '<input type="text"' },
```
Comment (aeschli) [inline src/test/completion.test.ts:72]: For the current attribute, the attribute should be in the list, e.g. to see the description or check that spelling is right

File: src/test/completion.test.ts [hunk 2]
```diff
@@ -69,15 +69,13 @@ suite('HTML Completion', () => {
...
 				{ label: 'tabindex', resultText: '<input tabindex="text"' }
 			]
 		});
 
 		testCompletionFor('<input type="text" |', {
 			items: [
 				{ label: 'style', resultText: '<input type="text" style="$1"' },
-				{ label: 'type', resultText: '<input type="text" type="$1"' },
```
Comment (aeschli) [inline src/test/completion.test.ts:80]: Use `notAvailable: true` to make sure the attribute proposal is not in the result

**Review by aeschli — COMMENTED** [review]
Please add a test case with an element that has multiple existing attributes and test completions also before the extension attributes

File: src/services/htmlCompletion.ts [hunk 3]
```diff
@@ -189,16 +189,30 @@ export class HTMLCompletion {
 			return result;
 		}
 
+		function getExistingAttributeNames(offset: number): { [attribute: string]: boolean } {
+			const existingAttributes = Object.create(null);
+			htmlDocument.findNodeBefore(offset).attributeNames.forEach(attribute => {
```
Comment (aeschli) [inline src/services/htmlCompletion.ts:194]: We already have the outcome of `htmlDocument.findNodeBefore(offset)` in `node`

File: src/services/htmlCompletion.ts [hunk 4]
```diff
@@ -189,16 +189,30 @@ export class HTMLCompletion {
...
 		function collectAttributeNameSuggestions(nameStart: number, nameEnd: number = offset): CompletionList {
 			let replaceEnd = offset;
 			while (replaceEnd < nameEnd && text[replaceEnd] !== '<') { // < is a valid attribute name character, but we rather assume the attribute name ends. See #23236.
 				replaceEnd++;
 			}
+			const existingAttributes = getExistingAttributeNames(nameStart);
+			const currentAttribute = text.substring(nameStart, nameEnd);
 			const range = getReplaceRange(nameStart, replaceEnd);
 			const value = isFollowedBy(text, nameEnd, ScannerState.AfterAttributeName, TokenType.DelimiterAssign) ? '' : '="$1"';
 			const seenAttributes = Object.create(null);
```
Comment (aeschli) [inline src/services/htmlCompletion.ts:209]: Maybe the existingAttributes can be added to the `seenAttributes`

**Review by aeschli — CHANGES_REQUESTED** [review]

**Comment by aeschli** [comment]
Thanks @sapphi-red for the work. I added some comments...

**Comment by sapphi-red** [comment]
Thank you for reviewing, I have fixed it!

**Review by aeschli — APPROVED** [review]

**Comment by aeschli** [comment]
Thanks a lot @sapphi-red !

**Pull request merged**